flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -173,6 +186,22 @@ public static StateBackend loadStateBackendFromConfig(
         }
     }
 
+    public static StateBackend loadWrappedStateBackendFromConfig(
+            ReadableConfig config, ClassLoader classLoader, @Nullable Logger logger)
+            throws IllegalConfigurationException, DynamicCodeLoadingException, IOException {
+
+        checkNotNull(config, "config");
+        checkNotNull(classLoader, "classLoader");
+
+        final StateBackend backend = loadStateBackendFromConfig(config, classLoader, logger);
+
+        if (config.get(CheckpointingOptions.ENABLE_STATE_CHANGE_LOG)) {
+            return loadChangelogStateBackend(backend, classLoader);
```

Comment:
I think the check whether `backend` is already delegating is missing here (`loadStateBackendFromConfig` can construct one using factory class).

Reply:
Do you mean `loadStateBackendFromConfig` would return a StateChangeLogStateBackend?

For now, all the wrapping logic is to construct a `StateChangeLogStateBackend` is through 
`loadChangelogStateBackend`

If we want to load through factory class (which we do not for now), we can add load through the factory 
through  `loadChangelogStateBackend` as well.

I think that is logically simpler and cleaner.

Reply:
> Do you mean loadStateBackendFromConfig would return a StateChangeLogStateBackend?

Yes.

Reply:
How do you think about my proposal above? If you do not like my proposal, maybe we can discuss this after I push the changed code.

Reply:
sync-up offline, will add a pre-check instead.